prov/sockets/src/sock_ep.c

```diff
@@ -1498,11 +1499,15 @@ struct fi_info *sock_fi_info(uint32_t version, enum fi_ep_type ep_type,
 {
 	struct fi_info *info;
 
-	info = fi_allocinfo();
+	info = fi_dupinfo(hints);
 	if (!info)
 		return NULL;
+	free(info->src_addr);
+	free(info->dest_addr);
```

Comment:
To be safe, please init info->src_addr and info->dest_addr to NULL after calling free.  src_addr won't matter, since we set it below.  But dest_addr might.